src/backend/src/main/kotlin/org/icpclive/overlay/Routing.kt

```diff
@@ -28,13 +29,52 @@ fun Route.configureOverlayRouting() {
     flowEndpoint("/mainScreen") { DataBus.mainScreenFlow.await() }
     flowEndpoint("/contestInfo") { DataBus.currentContestInfoFlow() }
     flowEndpoint("/runs") { DataBus.contestStateFlow.await().map { it.runsAfterEvent.values.sortedBy { it.time } } }
+    webSocket("/teamRuns") {
+        val teamIdStr = (incoming.receive() as? Frame.Text)?.readText()
+        if (teamIdStr.isNullOrBlank()) {
+            close(CloseReason(CloseReason.Codes.CANNOT_ACCEPT, "Invalid team id"))
+            return@webSocket
+        }
+        val teamId = teamIdStr.toTeamId()
+        sendJsonFlow(DataBus.contestStateFlow.await().map { state ->
+            state.runsAfterEvent.values.filter { it.teamId == teamId }
```

Comment:
I think this is a bit too time-consuming. Probably we should process events ourselves, ignoring events except known teams. 

Something like (didn't run or test to be compilable, just as main idea)
```
contestStateFlow.mapNotNull { (it.event as? RunEvent)?.newInfo }.runningFold(persistentMapOf<RunId, RunInfo>()) { acc, it ->
   if (it.teamId == teamId) acc.put(it.id, it) else if (it.runId in acc) acc.remove(it.runId) else acc
}.distintUntilChanged { a, b -> a === b }.map { it.value.sortedBy { it.time } }
```